Lecture-6/Task1/l6task1/Application.java

```diff
@@ -0,0 +1,139 @@
+            Coffee coffeeRobusta = new Coffee(coffeeType[i], coffeeSort[1], 2 * coffeePrice[i]);
+            coffees.add(coffeeArabica);
+            coffees.add(coffeeRobusta);
+        }
+        return coffees;
+    }
+
+    public static List<Product> createProducts(List<Pack> packs, List<Coffee> coffees) {
+        List<Product> products = new ArrayList<>();
+        for (Pack pack : packs) {
+            for (Coffee coffee : coffees) {
+                Product product = new Product(coffee, pack, getRandomProductBrand());
+                products.add(product);
+            }
+        }
+        return products;
+    }
+
+    public static void showCoffees(List<Coffee> coffees) {
+        for (Coffee coffee : coffees) {
+            System.out.println(coffee.toString());
+        }
+    }
+
+    public static void showPacks(List<Pack> packs) {
+        for (Pack pack : packs) {
+            System.out.println(pack.toString());
+        }
+    }
+
+    public static void showProducts(List<Product> products) {
+        for (Product product : products) {
+            System.out.println(product.toString());
+        }
+    }
+
+    public static int getSummaryProductsValue(List<Product> products) {
+        int summaryProductsValue = 0;
+        for (Product product : products) {
+            summaryProductsValue += product.getValue();
+        }
+        return summaryProductsValue;
+    }
+
+    public static int getSummaryProductsPrice(List<Product> products) {
+        int summaryProductPrice = 0;
+        for (Product product : products) {
+            summaryProductPrice += product.getPrice();
+        }
+        return summaryProductPrice;
+    }
+
+    public static void fillVan(List<Product> products, int money) {
+        Collections.shuffle(products);
+        List<Product> currentProducts = new ArrayList<>();
+        int currentVanValue = 0;
+        for (Product product : products) {
+            money -= product.getPrice();
+            if (money < 0) {
+                break;
+            }
+            currentVanValue += product.getValue();
+            if (currentVanValue > van.getMaxValue()) {
+                break;
+            }
+            currentProducts.add(product);
+        }
+        van.setProducts(currentProducts);
+    }
+
+    public static List<Product> getSortedProductsByPrice(List<Product> products) {
+        Collections.sort(products, new PriceProductComparator());
+        return products;
+    }
+
+    public static List<Product> findProducts(List<Product> products, String sort, int value) {
+        List<Product> sortedProducts = new ArrayList<>();
+        for (Product product : products) {
+            if (product.getCoffee().getSort().equals(sort) && product.getValue() == value) {
+                sortedProducts.add(product);
+            }
+        }
+        return sortedProducts;
+    }
+
+    public static void main(String[] args) {
```

Comment:
Current `Application` class becomes too long and difficult to read.
Let's extract methods to helper classes:
* ProductFactory - to create products
* ProductHelper - to search, sort , print products info
* ProductCalculator - all product calculations
* VanFiller - to check and fill in the van

Reply:
New classes are created. Factory pattern is applied.